File: go/string/7_reverse_int.go
package string

import "math"

// reverse 反转int32
func reverse(x int) int {

	// 实际上我们只要能拿到这个整数的 末尾数字 就可以了。
	var ret int

	for x != 0 {
		if x < math.MinInt32 / 10 || x > math.MaxInt32 / 10 {
			return 0
		}

		carry := x % 10
		x /= 10
		ret = ret * 10 + carry
	}
	return ret
}

File: rust/string/7_reverse_int.rs


pub fn reverse(x: i32) -> i32 {
    let mut x = x;
    let mut ret = 0;

    while x != 0 {
        if ret > i32::MAX / 10 || ret < i32::MIN / 10{
            return 0
        }

        let carry = x % 10;
        x /= 10;
        ret = ret *  10 + carry;
    }
    ret 
}